src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -1104,6 +1104,23 @@ public static void DCS_EnumerableCollection()
     }
 
     [Fact]
+    public static void DCS_BaseClassAndDerivedClassWithSameProperty()
+    {
+        var value = new DerivedClassWithSameProperty() { DateTimeProperty = new DateTime(100), IntProperty = 5, StringProperty = "TestString", ListProperty = new List<string>() };
+        value.ListProperty.AddRange(new string[] { "one", "two", "three" });
+        var actual = SerializeAndDeserialize<DerivedClassWithSameProperty>(value, "<DerivedClassWithSameProperty xmlns=\"http://schemas.datacontract.org/2004/07/SerializationTypes\" xmlns:i=\"http://www.w3.org/2001/XMLSchema-instance\"><DateTimeProperty>0001-01-01T00:00:00</DateTimeProperty><IntProperty>0</IntProperty><ListProperty i:nil=\"true\" xmlns:a=\"http://schemas.microsoft.com/2003/10/Serialization/Arrays\"/><StringProperty i:nil=\"true\"/><DateTimeProperty>0001-01-01T00:00:00.00001</DateTimeProperty><IntProperty>5</IntProperty><ListProperty xmlns:a=\"http://schemas.microsoft.com/2003/10/Serialization/Arrays\"><a:string>one</a:string><a:string>two</a:string><a:string>three</a:string></ListProperty><StringProperty>TestString</StringProperty></DerivedClassWithSameProperty>");
+
+        Assert.StrictEqual(value.DateTimeProperty, actual.DateTimeProperty);
+        Assert.StrictEqual(value.IntProperty, actual.IntProperty);
+        Assert.StrictEqual(value.StringProperty, actual.StringProperty);
+        Assert.NotNull(actual.ListProperty);
+        Assert.True(value.ListProperty.Count == actual.ListProperty.Count);
+        Assert.StrictEqual("one", actual.ListProperty[0]);
+        Assert.StrictEqual("two", actual.ListProperty[1]);
+        Assert.StrictEqual("three", actual.ListProperty[2]);
```

Comment:
Instead of hardcoding "one", "two", and "three" here again, should these be value.ListProperty[0], value.ListProperty[1], and value.ListProperty[2]?